setup.py

```diff
@@ -8,10 +8,6 @@
 import sys
 from typing import TYPE_CHECKING, Any
 
-if sys.version_info < (3, 8, 0):
-    sys.stderr.write("ERROR: You need Python 3.8 or later to use mypy.\n")
-    exit(1)
-
```

Comment:
This is an autofix for [ruff's UP036 rule](https://docs.astral.sh/ruff/rules/outdated-version-block/). In general, I think it's a useful rule, but here specifically I think it's better to `# noqa` it rather than deleting this block.